### PR TITLE
Web UI to test the library

### PR DESCRIPTION
**Pull request by bruno-garcia**
Those hacks with UseRouting didn't play well with the middlewares being added in the Web project.

Requires finding a way to read configuration while bootstrapping via WebHostBuilder or ConfigureServices/Configure but witout requiring such hacks:

```csharp
  services.Configure<RedirectrOptions>(section);
            services.Configure<RedirectrOptions>(c =>
            {
                if (c.BaseAddress is null)
                {
                    var baseAddress = _configuration?.GetValue<string>("URLS")?
                                          .Split(";")?
                                          .FirstOrDefault()
                                      // With TestServer 'URLS' isn't defined
                                      ?? "http://localhost";

                    c.BaseAddress = baseAddress;
                }
            });
```

### CODE REVIEW

File: src/Redirectr/ServiceCollectionExtensions.cs [hunk 1]
```diff
@@ -1,17 +1,22 @@
...
-        public static IServiceCollection AddRedirectr(this IServiceCollection serviceCollection)
+        public static IServiceCollection AddRedirectr(this IServiceCollection services)
         {
-            serviceCollection.TryAddSingleton<KeyGenerator>();
-            serviceCollection.TryAddSingleton<IRedirectrStore, InMemoryRedirectrStore>();
-            return serviceCollection;
+            services.TryAddSingleton<KeyGenerator>();
+            services.TryAddSingleton<IRedirectrStore, InMemoryRedirectrStore>();
+
+            services.TryAddEnumerable(
```
Comment (joaopgrassi) [inline src/Redirectr/ServiceCollectionExtensions.cs]: Why TryAddEnumerable?

Comment (joaopgrassi) [reply src/Redirectr/ServiceCollectionExtensions.cs]: Maybe `services.TryAddSingleton<IConfigureOptions<RedirectrOptions>, RedirectrOptionsSetup>();`

File: src/Redirectr.Web/appsettings.json [hunk 2]
```diff
@@ -6,8 +6,5 @@
       "Microsoft.Hosting.Lifetime": "Information"
     }
   },
-  "AllowedHosts": "*",
-  "Redirectr": {
-    "ShortUrlPath": "s"
```
Comment (joaopgrassi) [inline src/Redirectr.Web/appsettings.json:11]: Shouldn't we add a default then inside `RedirectrOptions`?

**Review by joaopgrassi — COMMENTED** [review]
left some comments, otherwise LGTM. 🚢

File: src/Redirectr/ServiceCollectionExtensions.cs (hunk 1 above)
Comment (bruno-garcia) [reply src/Redirectr/ServiceCollectionExtensions.cs]: To remove duplicate registrations

File: src/Redirectr.Web/appsettings.json (hunk 2 above)
Comment (bruno-garcia) [reply src/Redirectr.Web/appsettings.json:11]: Good catch! thanks

**Review by joaopgrassi — APPROVED** [review]

File: src/Redirectr/ApplicationBuilderExtensions.cs [hunk 3]
```diff
@@ -11,16 +10,14 @@
...
         public static IApplicationBuilder UseRedirectr(this IApplicationBuilder builder)
         {
             builder.UseEndpoints(endpoints =>
             {
                 var generator = endpoints.ServiceProvider.GetRequiredService<KeyGenerator>();
                 var options = endpoints.ServiceProvider.GetRequiredService<IOptions<RedirectrOptions>>().Value;
-                // TODO: Validate options such as, options.ShortenUrlPathTemplate is not null or empty
-                // BaseAddress exists/valid URL, ShortUrlPathPrefix is at least a slash or a valid path (/a/b/c/)
 
                 var shortenUrlPath = options.ShortenUrlPath;
```
Comment (tebeco) [inline src/Redirectr/ApplicationBuilderExtensions.cs]: All this block should probably be done in a `PostConfigure`
this code block : https://github.com/bruno-garcia/redirectr/blob/48d71f6fd86c4f07c813e3271a440f5366ac8264/src/Redirectr/ApplicationBuilderExtensions.cs#L22-L41

Comment (bruno-garcia) [reply src/Redirectr/ApplicationBuilderExtensions.cs]: I got rid of this. Rather used `Url` and did some of this stuff in PostConfigure

**Pull request merged**